### PR TITLE
Add field for maven coordinates

### PR DESCRIPTION
**Pull request by Azquelt**
We want to support storing resources in a maven repsository, so we need
to have a field for storing the coordinates.

### CODE REVIEW

File: client-lib/src/main/java/com/ibm/ws/repository/resources/writeable/RepositoryResourceWritable.java [hunk 1]
```diff
@@ -303,6 +303,15 @@
      */
     public void setLicenseId(String lic);
 
+    /**
+     * Set the maven coordinates
+     * <p>
+     * This is only set for resources which are stored in a maven repository.
+     * <p>
+     * Maven coordinates are returned as a string in the format {@code groupId:assetId:version}.
```
Comment (idlewis) [inline client-lib/src/main/java/com/ibm/ws/repository/resources/writeable/RepositoryResourceWritable.java]: Iffy javadoc. This is a setter not a getter

**Comment by idlewis** [comment]
@Azquelt No tests?

**Comment by Azquelt** [comment]
At some point I'll remember to do that _before_ you ask...

File: client-lib/src/main/java/com/ibm/ws/repository/resources/writeable/RepositoryResourceWritable.java (hunk 1 above)
Comment (Azquelt) [reply client-lib/src/main/java/com/ibm/ws/repository/resources/writeable/RepositoryResourceWritable.java]: I think it's ok. It's a getter for a field, but the person calling the getter needs to know that they shouldn't expect it to be set unless the resource is stored in a maven repository.

**Comment by Azquelt** [comment]
Test added, ready for review again.

**Review by idlewis — APPROVED** [review]

**Pull request merged**